### DISCUSSION TITLE
Octane Cache Issue

### DISCUSSION THREAD

**Discussion by amora2972** [spatie/laravel-translation-loader#176] [Q&A] [answered]
well, I am using this library in production with Octane, whenever I add a new language line, it's not reflected immediately to end users, a bit later it starts appearing on refresh and disappearing again on another refresh, after a long investigation I figured out the issue is related to serving my app using octane.

I am unsure if this is considered a bug, and if anyone has an idea on fixing it, please help me.

**Comment by erashdan** [comment, marked as answer]
I faced same issue it's worked with me by creating new listener 
FlushTranslatorCache

```php
<?php

namespace App\Listeners\Octane;

use Laravel\Octane\Events\RequestReceived;

class FlushTranslatorCache
{
    /**
     * Handle the event.
     */
    public function handle(RequestReceived $event): void
    {
        $translator = $event->app['translator'];

        $translator->setLocale($event->app->getLocale());

        $translator->setLoaded([]);
    }
}
```

after that you modify by adding to `config/octane.php` by adding a listener to RequestReceived event from octane
```php
     RequestReceived::class => [
            ...Octane::prepareApplicationForNextOperation(),
            \App\Listeners\Octane\FlushTranslatorCache::class, // <-- adding this
            ...Octane::prepareApplicationForNextRequest(),
        ],
```
